resources/services/kinesis/streams.go

```diff
@@ -0,0 +1,189 @@
+		Resolver:     fetchKinesisStreams,
+		Multiplex:    client.ServiceAccountRegionMultiplexer("kinesis"),
+		IgnoreError:  client.IgnoreAccessDeniedServiceDisabled,
+		DeleteFilter: client.DeleteAccountRegionFilter,
+		Options:      schema.TableCreationOptions{PrimaryKeys: []string{"arn"}},
+		Columns: []schema.Column{
+			{
+				Name:        "account_id",
+				Description: "The AWS Account ID of the resource.",
+				Type:        schema.TypeString,
+				Resolver:    client.ResolveAWSAccount,
+			},
+			{
+				Name:        "region",
+				Description: "The AWS Region of the resource.",
+				Type:        schema.TypeString,
+				Resolver:    client.ResolveAWSRegion,
+			},
+			{
+				Name:     "arn",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("StreamARN"),
+			},
+			{
+				Name:     "tags",
+				Type:     schema.TypeJSON,
+				Resolver: ResolveKinesisStreamTags,
+			},
+			{
+				Name:        "open_shard_count",
+				Description: "The number of open shards in the stream.  This member is required.",
+				Type:        schema.TypeBigInt,
+			},
+			{
+				Name:        "retention_period_hours",
+				Description: "The current retention period, in hours.  This member is required.",
+				Type:        schema.TypeBigInt,
+			},
+			{
+				Name:        "stream_arn",
+				Description: "The Amazon Resource Name (ARN) for the stream being described.  This member is required.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("StreamARN"),
+			},
+			{
+				Name:        "stream_creation_timestamp",
+				Description: "The approximate time that the stream was created.  This member is required.",
+				Type:        schema.TypeTimestamp,
+			},
+			{
+				Name:        "stream_name",
+				Description: "The name of the stream being described.  This member is required.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "stream_status",
+				Description: "The current status of the stream being described",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "consumer_count",
+				Description: "The number of enhanced fan-out consumers registered with the stream.",
+				Type:        schema.TypeBigInt,
+			},
+			{
+				Name:        "encryption_type",
+				Description: "The encryption type used",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "key_id",
+				Description: "The GUID for the customer-managed Amazon Web Services KMS key to use for encryption",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "stream_mode_details_stream_mode",
+				Description: "Specifies the capacity mode to which you want to set your data stream. Currently, in Kinesis Data Streams, you can choose between an on-demand capacity mode and a provisioned capacity mode for your data streams.  This member is required.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("StreamModeDetails.StreamMode"),
+			},
+		},
+		Relations: []*schema.Table{
```

Comment:
should we create a separate table for this field?

Reply:
As opposed to what? The field is an array within an array:

```
{
	StreamDescriptionSummary: &types.StreamDescriptionSummary{
		EnhancedMonitoring: []types.EnhancedMetrics{
			{ShardLevelMetrics: []types.MetricsName{types.MetricsNameAll},
		}
},
```

Reply:
agree, lets leave it as it is